### PR TITLE
Fix core detection when overkiz is ignored

### PR DESCRIPTION
**Pull request by vlebourl**

### CODE REVIEW

File: custom_components/tahoma/__init__.py [hunk 1]
```diff
@@ -222,11 +222,11 @@ def log_device(message: str, device: Device) -> None:
 async def _block_if_core_is_configured(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     overkiz_config_entries = hass.config_entries.async_entries("overkiz")
 
-    for overkiz_entry in overkiz_config_entries:
-        if (
-            entry.data[CONF_USERNAME] == overkiz_entry.data[CONF_USERNAME]
+    return any(
+        (
+            overkiz_entry.source != "ignore"
```
Comment (tetienne) [inline custom_components/tahoma/__init__.py:227]: Most of the time, there is always a constant, like [here](https://github.com/home-assistant/core/blob/494ef2f9b2e5feeb10dce94a02ee90a9a7430b6a/homeassistant/config_entries.py#L54)

**Comment by tetienne** [comment]
Good catch

File: custom_components/tahoma/__init__.py (hunk 1 above)
Comment (vlebourl) [reply custom_components/tahoma/__init__.py:227]: can be several things depending on how the user added the integration. For us, I guess one of those:
```py
SOURCE_USER = "user"
SOURCE_ZEROCONF = "zeroconf"
SOURCE_DHCP = "dhcp"
```

Comment (vlebourl) [inline custom_components/tahoma/__init__.py:227]:
```suggestion
            overkiz_entry.source in [SOURCE_USER, SOURCE_ZEROCONF, SOURCE_DHCP]
```

Comment (vlebourl) [reply custom_components/tahoma/__init__.py:227]: Is that ok with you?

File: custom_components/tahoma/__init__.py [hunk 2]
```diff
@@ -222,11 +227,11 @@ def log_device(message: str, device: Device) -> None:
 async def _block_if_core_is_configured(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     overkiz_config_entries = hass.config_entries.async_entries("overkiz")
 
-    for overkiz_entry in overkiz_config_entries:
-        if (
-            entry.data[CONF_USERNAME] == overkiz_entry.data[CONF_USERNAME]
+    return any(
+        (
+            overkiz_entry.source in [SOURCE_USER, SOURCE_ZEROCONF, SOURCE_DHCP]
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:232]:
```suggestion
            overkiz_entry.source in [SOURCE_USER, SOURCE_ZEROCONF, SOURCE_DHCP]
```

When you set up an entry (even when clicked via zeroconf/dhcp), doesn't it always use source user? Wouldn't this current change detect if a hub has been discovered but not set up as well?

Comment (vlebourl) [reply custom_components/tahoma/__init__.py:232]: From my testing, hubs that haven't been set up ar not in `hass.config_entries.async_entries("overkiz")`. And the source is kept as `zeroconf` (or `dhcp` I guess but untested) if you set up a hub that was auto discovered. I think to get `user`, you need to actively select "Add Integration" then "No, set up another instance of Overkiz (by Somfy)".

Comment (vlebourl) [reply custom_components/tahoma/__init__.py:232]: image

Comment (iMicknl) [reply custom_components/tahoma/__init__.py:232]: Thanks! It is always a bit vague how that behaves, thanks for double checking.

**Review by iMicknl — APPROVED** [review]

File: custom_components/tahoma/__init__.py (hunk 1 above)
Comment (tetienne) [reply custom_components/tahoma/__init__.py]: @vlebourl Why did you change your mind?

**Pull request merged**